kernel/aster-nix/src/fs/inode_handle/dyn_cap.rs

```diff
@@ -61,6 +64,101 @@ impl InodeHandle<Rights> {
         }
         self.0.readdir(visitor)
     }
+
+    pub fn position(&self) -> Result<i64> {
+        Ok(self.0.offset() as i64)
+    }
+
+    pub fn test_range_lock(&self, lock: &mut RangeLock) -> Result<()> {
```

Comment:
I think we should move these new code into `mod.rs`, and the code in `dyn_cap.rs` just do some permission check and call the functions in `mode.rs`, this is the design of `InodeHandle`.